sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -146,7 +146,7 @@ trait CodegenSupport extends SparkPlan {
       if (outputVars != null) {
         assert(outputVars.length == output.length)
         // outputVars will be used to generate the code for UnsafeRow, so we should copy them
-        outputVars.map(_.copy())
+        outputVars.map(_.copy()).toBuffer
```

Comment:
can we do something like:
```
val res = outputVars.map(_.copy())
res  match {
 case x: Stream => x.force
 case o => o
}
```

in order to avoid doing the `toBuffer` operation when not needed (please use more meaningful names than in my example in case it is feasible :) )?

cc @hvanhovell who I remember fixed a similar issue with `Stream`s.